src/hotspot/cpu/x86/x86.ad

```diff
@@ -5775,6 +5775,46 @@ instruct evminmaxFP_reg_eavx(vec dst, vec a, vec b, vec atmp, vec btmp, kReg ktm
   ins_pipe( pipe_slow );
 %}
 
+// --------------------------------- Signum ---------------------------
+
+instruct signumF_reg(regF dst, regF zero, regF one, rFlagsReg cr) %{
+  match(Set dst (SignumF dst (Binary zero one)));
+  effect(KILL cr);
+  format %{ "signumF $dst, $dst" %}
+  ins_encode %{
+    Label exit;
+
```

Comment:
Do you mean `instruct signumF_reg`? Please explain your reasoning for the proposed move.

Reply:
I am referring to moving the instruction sequence into a macro assembly routine for better code sharing, one should be enough for both float and double type, please refer to following snippet for more detail.
https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/x86/x86.ad#L5672

Reply:
Hi @jatin-bhateja your original request was quite unclear. The highlighting of the code was incorrect and the use of 'this' made it unclear what you were suggesting and you did not explain a reason for the suggestion. 

> Can you kindly move **this** into a macro assembly routine.

Seeing "better code sharing" as the reason it's unclear to me if the proposed refactor results in any meaningful benefit for the work involved. It's also unclear if there are any hard code structure or style issues that would necessitate the move.

The refactor you suggest calls for the instruction sequences here to be moved into a src/hotspot/cpu/x86/c2_MacroAssembler_x86.cpp function that is then called here. Inside that function there is a selection between float and double but otherwise stays the same except one is formatted .AD and one is .CPP/.HPP. 

If I am doing this refactor for `instruct signumF_reg` and `instruct signumD_reg` to cut down on code duplication, I see that as close to a wash for saving of code lines and effort involved. I also think this would make the code more complex to read.

However if there are strong opinions as to why this is necessary I'd be happy to hear them.

Reply:
I think it'd be easier to understand with a signum macro in MacroAssembler. Less code duplication is good, obviously. And it's potentially useful elsewhere. Win-win-win.
We've already factored out C2 instruction sequences for things like vabsnegd  into c2_MacroAssembler_x86.

Reply:
I wave the white flag. 

Thanks for the review and suggestions @theRealAph and @jatin-bhateja.

In an effort to eradicate the 5 lines of duplication and with an eye to potential usefulness in the future I've created a twice called `C2_MacroAssembler::signum_fp` using 36 lines of code and modification of c2_MacroAssembler_x86.cpp c2_MacroAssembler_x86.hpp.